### ISSUE TITLE
Internationalize and bulletproof file, project, etc. naming form fields 

### ISSUE THREAD

**Issue by JonFerraiolo** [maqetta/maqetta#3828]
@peller @billreed63 @jhpedemonte 

For various form fields (e.g., file name, project name, project template name, theme name), JavaScript doesn't offer convenient regular expression features to allow for any international alphanumeric (basically, I'm looking for a `\w` except one that isn't restricted to ASCII). Java and other languages offer `\p{L}` to mean "any alpha" but that construct is not supported in JavaScript regular expressions. Also, Dojo has nothing in this space.

Instead, for JavaScript, one has to manually list the various Unicode code points in a regular expression that represent an alpha. There are various things available on the Web, but we need to be careful about IP issues, so it might be best for someone to figure things out from scratch looking at Unicode specs.

**Comment by jhpedemonte** [comment]
One option:  for the login fields, at least, Orion doesn't do client side checking.  Rather, the form values are submitted and the server does the checking (which it would have to anyway, regardless of client-side checking).  If validation fails, an error is returned to the client.

**Comment by peller** [comment]
What is it we're trying to bulletproof?  Is this just where we need to avoid certain characters like slash and dot?  I think in many cases, alphanumeric is already too restrictive, even for just ASCII, but we could extend what we have now to allow unicode alphabets by just adding the unicode range above ascii.  Do we really need to pick and choose from all those alphabets?

@jhpedemonte has a point. Server-side validation is ultimately the right thing to do, especially from a REST point of view.

**Comment by JonFerraiolo** [comment]
> Server-side validation is ultimately the right thing to do, especially from a REST point of view.

Yes, we need to do server-side validation, but I don't think we want to implement `cmd/validateProjectName` and potentially several other REST calls, and do REST calls with every keystroke in order to update the UI on ValidateTextBox, when a simple regex would solve the client-side validation problem. We just need to make sure that the client-side regex doesn't allow any characters that will be rejected by the server-side validator.

**Comment by peller** [comment]
I'm not suggesting implementing a server-side REST service for validation, just making sure our existing REST services which take inputs for project name do the necessary validation and/or bulletproof there.
